config/routes.rb

```diff
@@ -1,4 +1,4 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
   get 'companies/:company_id/shipments', to: 'shipments#index'
+  get 'companies/:company_id/shipments/:shipment_id', to: 'shipments#show'
```

Comment:
Why do you not follow the Rails convention?
```suggestion
  get 'companies/:company_id/shipments/:id', to: 'shipments#show'
```

Reply:
I might use request spec instead of controller specs, and now studying how to write it.
I forgot to use that convention! thanks for reminder!

Reply:
Hi, Tomo
I found that if I just replace `:shipment_id` with `:id`, it'll get 404 response 
I just looked for come infos here: https://guides.rubyonrails.org/v3.2.0/routing.html
I think I could just modify `:company_id` to `:id`, and it looks normal in my localhost

Reply:
If you update here, you should update your controller as well. I think that's why 404 (not found) return.
Also, why did you look up documents for Rails 3.2 😆 

And it's not correct to change `:company_id` to `:id`, that's not for RESTful.

Can you try to use `resources` to create routes and run `rake routes` to see their difference

Reply:
Oh!!! So that's because I set the params to find shipment!!@@
Thank you, I finally know get your point! :)